src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -377,6 +384,99 @@ public void WriteValue(string value)
             this.FlushIfBufferThresholdReached();
         }
 
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            int chunkSize = bufferWriter.FreeCapacity;
+
+            for (int i = 0; i < value.Length; i += chunkSize)
+            {
+                int remainingChars = Math.Min(chunkSize, value.Length - i);
+                ReadOnlySpan<char> chunk = value.Slice(i, remainingChars);
+
+                if (NeedsEscaping(chunk) != -1)
+                {
+                    WriteEscapedStringChunk(chunk);
+                }
+                else
+                {
+                    WriteStringChunk(chunk);
+                }
+
+                // Flush the buffer if needed
+                this.FlushIfBufferThresholdReached();
+            }
+        }
+
+        private int NeedsEscaping(ReadOnlySpan<char> value)

```

Comment:
since this returns an index, I think you should call it `GetIndexOfFirstCharToEscape`